core/primitives/src/utils.rs

```diff
@@ -465,10 +469,27 @@ where
     Serializable(object)
 }
 
+/// Derives `AccountId` from `PublicKey``.
+/// If the key type is ED25519, returns hex-encoded copy of the key.
+pub fn derive_near_implicit_account_id(public_key: &PublicKey) -> AccountId {
+    match public_key.key_type() {
+        KeyType::ED25519 => hex::encode(public_key.key_data()).parse().unwrap(),
+        _ => unimplemented!(),
```

Comment:
Is that method used in production code or is it only for tests? I'm a bit weary of triggering this panic here if someone provides some other key type.

Reply:
Currently it is used in tests only, and one usage in the mirror tool.
I changed the signature to accept `ED25519PublicKey` only.